src/ResourceManager/Compute/Commands.Compute/Strategies/Network/PublicIPAddressStrategy.cs

```diff
@@ -45,8 +47,24 @@ static class PublicIPAddressStrategy
                     PublicIPAllocationMethod = allocationMethod,
                     DnsSettings = new PublicIPAddressDnsSettings
                     {
-                        DomainNameLabel = domainNameLabel,                       
+                        DomainNameLabel = domainNameLabel.Value,                       
                     }
                 });
+
+        public static async Task FindDomainNameLabelAsync(
+            Mutable<string> domainNameLabel,
+            string name,
+            string location,
+            IClient client)
+        {
+            if (domainNameLabel.Value == null)
+            {
+                var networkClient = client.GetClient<NetworkManagementClient>();
+                do
+                {
+                    domainNameLabel.Value = (name + '-' + Guid.NewGuid().ToString().Substring(0, 6)).ToLower();
+                } while ((await networkClient.CheckDnsNameAvailabilityAsync(location, domainNameLabel.Value)).Available != true);
```

Comment:
Instead of `!= true`, just throw a `!` out front?

Reply:
This is `bool?` so I would like to have explicit syntax.